internal/dag/httpproxy_processor.go

```diff
@@ -1343,6 +1334,53 @@ func (p *HTTPProxyProcessor) computeSecureVirtualHostAuthorization(validCond *co
 	return true
 }
 
+func computeVirtualHostRateLimitPolicy(proxy *contour_api_v1.HTTPProxy, rateLimitService *contour_api_v1alpha1.RateLimitServiceConfig, validCond *contour_api_v1.DetailedCondition) *RateLimitPolicy {
+	rlp := &RateLimitPolicy{}
+	var err error
+
+	// Set the local rateLimit policy
+	if proxy.Spec.VirtualHost.RateLimitPolicy != nil && proxy.Spec.VirtualHost.RateLimitPolicy.Local != nil {
```

Comment:
can we return early if `proxy.Spec.VirtualHost.RateLimitPolicy==nil`?

Reply:
No we cannot because the default global ratel limit policy might be set